src/librustc_codegen_llvm/common.rs

```diff
@@ -260,9 +260,23 @@ impl ConstMethods<'tcx> for CodegenCx<'ll, 'tcx> {
                     Some(GlobalAlloc::Memory(alloc)) => {
                         let init = const_alloc_to_llvm(self, alloc);
                         if alloc.mutability == Mutability::Mut {
-                            self.static_addr_of_mut(init, alloc.align, None)
+                            let value = self.static_addr_of_mut(init, alloc.align, None);
+                            if !self.sess().fewer_names() {
```

Comment:
to deduplicate this you could make it a closure that you just call at these two sites

Reply:
Thanks Oliver.  I'll wait to see if we can unblock this issue https://github.com/rust-lang/rust/pull/69155#discussion_r384973041 before I spend any more time on this.

Reply:
I don't think you need a closure, you can just bind the result of the `if`-`else` instead of having `let value` in both sides.

Reply:
01b7e647fec04d9b086cb8cc7430f290befa8d77 removes the duplication